x-pack/plugins/fleet/server/routes/agent/helpers.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SavedObjectsClientContract } from '@kbn/core/server';
+
+import { downloadSourceService, agentPolicyService } from '../../services';
+import type { Agent, AgentPolicy } from '../../types';
+
+export const getSourceUriForAgent = async (soClient: SavedObjectsClientContract, agent: Agent) => {
```

Comment:
nit: it would be nice to call this file something more meaningful e.g. source_uri_utils.ts